File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02730/p02730_num8_parsed.cpp
#include <iostream>
#include <string>
#include <algorithm>

bool isPalindrome(const std::string& s) {
    int n = s.length();
    for (int i = 0; i < n / 2; i++) {
        if (s[i] != s[n - 1 - i]) {
            return false;
        }
    }
    return true;
}

bool isStrongPalindrome(const std::string& s) {
    int n = s.length();
    std::string firstHalf = s.substr(0, (n - 1) / 2);
    std::string secondHalf = s.substr((n + 3) / 2 - 1);
    return isPalindrome(s) && isPalindrome(firstHalf) && isPalindrome(secondHalf);
}

int main() {
    std::string S;
    std::cin >> S;

    if (isStrongPalindrome(S)) {
        std::cout << "Yes" << std::endl;
    } else {
        std::cout << "No" << std::endl;
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02730/p02730_num8_parsed.py
def is_palindrome(s):
    return s == s[::-1]

def is_strong_palindrome(s):
    n = len(s)
    first_half = s[:((n-1)//2)]
    second_half = s[((n+3)//2)-1:]
    
    return is_palindrome(s) and is_palindrome(first_half) and is_palindrome(second_half)

# Main code
s = input()
if is_strong_palindrome(s):
    print("Yes")
else:
    print("No")
